pytradfri/command.py

```diff
@@ -32,8 +32,8 @@ def __init__(
         self._err_callback = err_callback
         self._observe = observe
         self._observe_duration = observe_duration
-        self._raw_result: str | None = None
-        self._result: str | None = None
+        self._raw_result: dict | str | None = None
```

Comment:
When is it a `dict` and when is it a `str`? Should we perhaps add a line or two to document this?

Reply:
Yeah. It's a string when generating the psk and when getting the endpoints since those need custom parsing. I can add a comment.

I have follow up work planned that will narrow down the result types.

Reply:
It can also be a list[dict].

Reply:
Are you referring to when multiple commands are sent? Each command result is processed individually.

Reply:
I am referring to what I have seen (in debugger) being returned from aiocoap and assigned.

Reply:
I believe the best way to reproduce it, is to make several events in the gateway at the same time using e.g. 2 ikea buttons and 2 lights. I do not believe you can produce it by sending commands.

Reply:
Yes, you're right, the command to get devices, groups etc, returns a list. I'll fix this.

Reply:
when correcting please remember it can also be an empty list.